corehq/apps/translations/app_translations/download.py

```diff
@@ -19,39 +19,60 @@
     get_module_sheet_name,
     get_modules_and_forms_row,
 )
+from corehq.apps.translations.generators import EligibleForTransifexChecker
 
 
-def get_bulk_app_single_sheet_by_name(app, lang):
+def get_bulk_app_single_sheet_by_name(app, lang, eligible_for_transifex_only=False):
+    checker = EligibleForTransifexChecker(app)
+
     rows = []
     for module in app.modules:
+        if eligible_for_transifex_only and checker.exclude_module(module):
```

Comment:
can extract a private method for better readability
```
def include_module_or_form(module_or_form):
    return eligible_for_transifex_only and checker.exclude_module(module)
```
and then use that here and for the form.

^ the logic above is not accurate but just an idea

Reply:
Similarly for module_row. You can also consider filter all module_rows and then just appending them all at once.

Reply:
May be for form_media as well. Basically extracting out the blocks of logic into methods so that the naming can help understand them quickly and the logic is contained in a method for better absorption of what all is happening in this long method.